cmd/pebble/main.go

```diff
@@ -18,11 +18,22 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/canonical/pebble/client"
 	"github.com/canonical/pebble/internals/cli"
 )
 
 func main() {
-	if err := cli.Run(nil); err != nil {
+	cliOptions := &cli.RunOptions{
+		ClientConfig: &client.Config{
+			Socket:        os.Getenv("PEBBLE_SOCKET"),
+			BaseURL:       os.Getenv("PEBBLE_BASEURL"),
```

Comment:
@benhoyt I propose we consider exposing PEBBLE_BASEURL. It supports all the ```url.URL``` bits we need:

```http://localhost:8888" - HTTP on 8888
```https://localhost:8433" - HTTPS on 8433

Reply:
I like the idea of a single `PEBBLE_BASEURL` which can include `user:pass` (alternatively we could have separate `PEBBLE_USERNAME` and `PEBBLE_PASSWORD` env vars). However, in the past Gustavo's pushed back against putting credentials in env vars altogether. Maybe we need a special file in the Pebble client home dir (`$XDG_CONFIG_HOME/pebble/credentials.yaml` or something -- see also `cliStatePath`). I think this is worth discussing with Gustavo.

Reply:
This whole topic of how to supply the basic auth credentials, and also how the pairing process works between the client and the server is moved for discussions next cycle. I will close this topic for now. PEBBLE_BASEURL works, but ignored the user:pass part.